app/src/main/java/com/postpc/tenq/ui/ui_helpers/ShareAlertDialog.java

```diff
@@ -0,0 +1,59 @@
+package com.postpc.tenq.ui.ui_helpers;
+
+import android.app.AlertDialog;
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.postpc.tenq.R;
+
+import net.glxn.qrgen.android.QRCode;
+
+public class ShareAlertDialog {
+
+
+    public static void share(Context context, String curRoomId) {
+        LayoutInflater inflater = (LayoutInflater) context.getSystemService( Context.LAYOUT_INFLATER_SERVICE );
+        View view = inflater.inflate(R.layout.activity_share, null);
+
+        String roomId = curRoomId;
+
+        generateQrCode(view, roomId);
+
+        AlertDialog alertDialog = new AlertDialog.Builder(context)
+                .setView(view).create();
+        alertDialog.show();
+
+        // cancel button
+        view.findViewById(R.id.txt_cancel).setOnClickListener(cancel -> alertDialog.dismiss());
+
+        // copy link button
+        view.findViewById(R.id.copy_link).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                copyDataToClipBoard();
+            }
+
+            private void copyDataToClipBoard() {
+                ClipData clipData = ClipData.newPlainText("text", roomId);
+                ClipboardManager clipboardManager = (ClipboardManager)  context.getSystemService(Context.CLIPBOARD_SERVICE);
+                clipboardManager.setPrimaryClip(clipData);
+                Toast.makeText(context.getApplicationContext(), "Data Copied to Clipboard", Toast.LENGTH_SHORT).show();
```

Comment:
This line is for debugging, right? Add a todo to remove it when the app is ready?